build.gradle

```diff
@@ -0,0 +1,354 @@
+                      "picard.reference.ExtractSequences", "picard.analysis.QualityScoreDistribution", "picard.vcf.RenameSampleInVcf", "picard.sam.ReorderSam",
+                      "picard.sam.ReplaceSamHeader", "picard.sam.RevertSam", "picard.sam.RevertOriginalBaseQualitiesAndAddMateCigar", "picard.sam.SamFormatConverter",
+                      "picard.sam.SamToFastq", "picard.util.ScatterIntervalsByNs", "picard.sam.SortSam", "picard.vcf.SortVcf", "picard.sam.SplitSamByLibrary",
+                      "picard.vcf.UpdateVcfSequenceDictionary", "picard.vcf.VcfFormatConverter", "picard.illumina.MarkIlluminaAdapters", "picard.vcf.SplitVcfs",
+                      "picard.sam.ValidateSamFile", "picard.sam.ViewSam", "picard.vcf.VcfToIntervalList"]
+
+task documentInit(dependsOn: 'classes') {
+    commandLineHtmlDir.delete()
+    commandLineHtmlDir.mkdirs()
+}
+
+task documentAll
+
+task documentStandardOptions(dependsOn: 'documentInit', type: JavaExec) {
+    main = 'picard.cmdline.CreateHtmlDocForStandardOptions'
+    classpath = sourceSets.main.runtimeClasspath
+    standardOutput = new FileOutputStream(standardOptionsFile)
+    documentAll.dependsOn "documentStandardOptions"
+}
+
+ext.lastDocTask = null
+ext.currentDocTask = null
+
+commandClasses.each { mainClass ->
+    currentDocTask = task "document_${mainClass}"(dependsOn: 'documentInit', type: JavaExec) {
+        main = 'picard.cmdline.CreateHtmlDocForProgram'
+        classpath = sourceSets.main.runtimeClasspath
+        args mainClass
+        standardOutput = new FileOutputStream(new File(commandLineHtmlDir, mainClass.substring(mainClass.lastIndexOf(".") + 1) + ".html"))
+        doLast {
+            usageFile.append("{% include ${mainClass.substring(mainClass.lastIndexOf(".") + 1) + ".html"} %}")
+            usageFile.append(System.getProperty("line.separator"))
+            sidebarFile.append("<li><a href=\"command-line-overview.html#${mainClass.substring(mainClass.lastIndexOf(".") + 1)}\">${mainClass.substring(mainClass.lastIndexOf(".") + 1)}</a>")
+            sidebarFile.append(System.getProperty("line.separator"))
+        }
+    }
+    if (lastDocTask != null) currentDocTask.dependsOn lastDocTask
+    lastDocTask = currentDocTask
+    documentAll.dependsOn lastDocTask
+}
+//end generate documentation
+
+task wrapper(type: Wrapper) {
+    description = "Regenerate the gradle wrapper"
+    gradleVersion = '2.13'
+}
+
+task createMetricsDoc(dependsOn: ['documentInit', 'classes'], type: Javadoc){
+    source = sourceSets.main.allJava
+    classpath = sourceSets.main.runtimeClasspath
+    destinationDir = file('dist/html')
+    options.doclet = 'picard.util.MetricsDoclet'
+    options.docletpath = sourceSets.main.runtimeClasspath.asType(List)
+    documentAll.dependsOn "createMetricsDoc"
+}
+
+task javadocJar(type: Jar, dependsOn: javadoc) {
```

Comment:
should this depend on `documentAll` instead of javadoc?  It might be best to configure the default javadoc instead of using a custom task here.  We can always revise later though if needed.